source/PlayerInfo.cpp

```diff
@@ -2016,6 +2018,58 @@ void PlayerInfo::SetTravelDestination(const Planet *planet)
 
 
 
+void PlayerInfo::QueueTeleport(const Planet *teleportPlanet, bool _flagshiponly)
+{
+	teleportDestination = teleportPlanet;
+	flagshiponly = _flagshiponly;
+}
+
+
+
+void PlayerInfo::DoQueuedTeleport()
+{
+	if(teleportDestination && planet)
+	{
+		flagship->SetSystem( teleportDestination->GetSystem());
+		flagship->SetPlanet(teleportDestination);
+		if(!flagshiponly)
+		{
+			for(const shared_ptr<Ship> &ship : ships)
+			{
+				if(!ship->IsParked() && !ship->IsDestroyed() && !ship->IsDisabled() && ship->GetSystem() == system)
+				{
+					ship->SetSystem( teleportDestination->GetSystem());
+					ship->SetPlanet(teleportDestination);
+				}
+			}
+		}
+		system = teleportDestination->GetSystem();
+		planet = teleportDestination;
+		teleportationStatus = TELEPORTING;
+		teleportDestination = nullptr;
+	}
+	if(!planet && teleportDestination)
+	{
```

Comment:
No need for braces here either
Also just put if(teleportDestination) in an if that has 
if(planet)
  Lines 2033 to 2049
else
  This thing herr

Reply:
dont get it please explain

Reply:
Its a one line code so you dont need braces
Also I was suggesting only evaluating teleportDestination once, in a bigger if encompassing the two ifs, and have the first if just be checking planet; with the second being the else

Reply:
oh, ok

Reply:
done